app/services/stories_api/v3/endpoints/story.rb

```diff
@@ -35,7 +36,12 @@ def get
         end
 
         def patch
-          story = user.user_sets.custom_find(params[:id])
+          story = if RecordSchema.roles[@user.role.to_sym].try(:admin)
```

Comment:
Could do, but though out the api, this seems to be the way to check a users role.  What does @EdwinRozario think?